ui/src/app/utils/search/filter-nodes.ts

```diff
@@ -0,0 +1,177 @@
+import FilterHandlers from "./filter-handlers";
+import type { Filters, FilterValue, PrefixedFilter } from "./filter-handlers";
+
+export type GetValue<N> = (
+  node: N,
+  filter: string
+) => FilterValue | FilterValue[] | null;
+
+export default class FilterNodes<N, PK extends keyof N> extends FilterHandlers {
```

Comment:
These filter utilities are also now in a class so that the generics and underlying filters for `prefixedFilters` (workload annotations) are set up in one place.